lib/Parser/Context/Variables.pm

```diff
@@ -26,15 +26,12 @@ sub init {
   $self->{dataName} = 'variables';
   $self->{name} = 'variable';
   $self->{Name} = 'Variable';
-  $self->{namePattern} = qr/[a-zA-Z][a-zA-Z0-9]*/;
+  $self->{namePattern} = qr/\w+/;
   $self->{tokenType} = 'var';
-  $self->{precedence} = 5;
+  $self->{precedence} = 10;
```

Comment:
The precedence change is "black magic" for me. Maybe for the record put an explanation here or in the file.

Reply:
I've added a comment to the file.

The change is because I moved the number pattern from being before all the specifically named items (functions, variables, operators, etc.) too after them, so that names can include leading numbers (e.g., a string `1st-degree` would be allowed now, but not before).  I set the number precedence to 5 (standard tokens are 0), to get a little room to insert others in between, if needed in the future.  Then I changed this precedence to 10 to put it after numbers (with a little room for more in-between).  This does mean that variable names can't start with a number.  One can set this to between 0 and 5 to allow variables that start with a leading number, if that is desired.

Reply:
Thanks for explaining/.